src/libraries/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -165,6 +166,37 @@ public unsafe BitArray(bool[] values)
                     }
                 }
             }
+            else if (AdvSimd.Arm64.IsSupported)
+            {
+                Vector128<byte> elementMask = Vector128.Create(0x01020408_10204080).AsByte();
+                fixed (bool* ptr = values)
+                {
+                    for (; (i + Vector128<byte>.Count * 2) <= values.Length; i += Vector128<byte>.Count * 2)
+                    {
+                        // Same logic as SSE2 path, however we lack MoveMask (equivalent) instruction
+                        // As a workaround, mask out the relevant bit after comparison
+                        // and combine by ORing all of them together (In this case, adding all of them does the same thing)
+                        Vector128<byte> lowerVector = AdvSimd.LoadVector128((byte*)ptr + i);
+                        Vector128<byte> lowerIsFalse = AdvSimd.CompareEqual(lowerVector, Vector128<byte>.Zero);
```

Comment:
Move `Vector128<byte>.Zero` out of the loop (save to a variable). Jit doesn't support loop code hoisting for SIMD yet.

Reply:
Given https://github.com/dotnet/runtime/pull/33749#issuecomment-601447809 (`cmeq (register) when one of the operands is Vector128<T>.Zero will be replaced with cmeq (zero) when we support this optimization`),
do you think it would be okay to leave it as-is, at least for the ARM path?

Reply:
I would follow advise of @EgorBo and hoisted it out of the loop - with comment saying that it would be not be needed as soon as https://github.com/dotnet/runtime/issues/33972 is implemented.

You can also put the same comment on the issue thread.

Reply:
Done, will comment on the issue once the PR is merged.